src/main/groovy/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.groovy

```diff
@@ -151,4 +154,16 @@ public class Utils {
             return false
         }
     }
+
+    @Whitelisted
+    static void setSkippedStage(String stageName) {
+        CpsThread thread = CpsThread.current()
+        CpsFlowExecution execution = thread.execution
+
+        FlowNode currentNode = execution.currentHeads.find { n ->
+            n?.displayName?.equals(stageName)
+        }
+
+        currentNode.actions.add(new NotExecutedNodeAction())
```

Comment:
JENKINS-26522 is the way to do something like this, NotExecutedNodeAction has a very specific use case and it's not this, and this would break expected behavior.